File: client-app/src/app/models/propertyType.ts
import { IProperty } from "./Property";
import { IAmenities } from "./amenities";

export interface IPropertyType {
    id: string;
    name: string;
    description: string;
    location: string;
    image?: Image;
    properties?: IProperty[] | null;
    amenities?: IAmenities[] | null;
}

export interface IPropertyTypeFormValues extends Partial<IPropertyType> {}

export class PropertyTypeFormValues implements IPropertyTypeFormValues {
  id?: string = undefined;
  name: string = '';
  description: string = '';
  location: string = '';
  properties?: IProperty[] = undefined;
  amenities?: IAmenities[] = undefined;
}

export interface Image {
    id: string;
    url: string;
    isMain: boolean;
} 

File: Domain/PropertyType.cs
using System;
using System.Collections.Generic;

namespace Domain
{
    public class PropertyType
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string Location { get; set; }
        public virtual Photo Image { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
        public DateTime? DeletedAt { get; set; }

        public virtual ICollection<Amenity> Amenities { get; set; }

        // Navigation Properties
        public virtual ICollection<Property> Properties { get; set; }
        public virtual ICollection<Photo> Photos { get; set; }
        // public object Amenities { get; set; }
        public virtual ICollection<Amenity> Amenities { get; set; }
        // public virtual ICollection<Transaction> Transactions { get; set; } // Old

    }
}